src/simulator/planner_simulator.py

```
@@ -0,0 +1,95 @@
+            privacy budget consumption for this simulation.
+        """
+        self._halo = halo
+        self._params = params
+        self._privacy_tracker = privacy_tracker
+        self._modeling_strategy = modeling_strategy
+
+    def fit_model(self, budget: PrivacyBudget) -> None:
+        """Fits a model using data_set, parameters and modeling_strategy.
+
+        budget:  A PrivacyBudget object specifying how much privacy budget
+          is to be consumed when fitting this model.
+        """
+        self._model = self._modeling_strategy.fit(self._halo, self._params, budget)
+
+    def true_reach_by_spend(
+        self, spends: List[float], max_frequency: int = 1
+    ) -> ReachPoint:
+        """Returns the true reach obtained for a given spend vector.
+
+        Args:
+            spends:  The hypothetical amount spend vector, equal in length to
+              the number of publishers.  spends[i] is the amount that is
+              spent with publisher i.
+            max_frequency:  The maximum frequency for which to report reach.
+        Returns:
+            A ReachPoint representing the true reach that would have been
+            obtained for this spend allocation.
+        """
+        return self._halo.true_reach_by_spend(spends, max_frequency)
+
+    def modeled_reach_by_spend(
+        self, spends: List[float], max_frequency: int = 1
+    ) -> ReachPoint:
+        """Returns the reach estimated via the reach surface model.
+
+        Args:
+            spends:  The hypothetical amount spend vector, equal in length to
+              the number of publishers.  spends[i] is the amount that is
+              spent with publisher i.
+            max_frequency:  The maximum frequency for which to report reach.
+        Returns:
+            A ReachPoint representing the true reach that would have been
+            obtained for this spend allocation.
+        """
+        return self._model.by_spend(spends, max_frequency)
```

Maybe throw an error if this function is called fit_model?

---

This would make sense if the caller might want to catch the exception and work around it, but my feeling is that the best course of action would be just to crash.  What do you think?

---

sure, your call. Just wanted to point out in case you had missed that use case